Managing Teams of Users for Policies

dwhogan opened the pull request:
## Links to Jira Tickets
 
https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-828

## Description of Changes

As a BioHub system administrator, I want to create teams of users that can be assigned to access policies, so that I can minimize the workload of managing individual user access across multiple policies.

Acceptance Criteria
1. Manage Teams Container  
WHEN administrators access the policy management page,  
  THEN a "Manage Teams" container should be present on the same page where policies are created.

2. Team Creation  
WHEN a BioHub administrator wants to create a new team,
THEN the system should allow creating a team with:
a team name,
an optional description, and
a list of users selected from the system user table, excluding SYSTEM and DATABASE users.

3. Editing Teams  
WHEN a BioHub administrator wants to modify an existing team,  
  THEN they should be able to update the team name, description, and membership.

4. Deleting Teams  
WHEN a BioHub administrator decides to remove a team,  
  THEN the system should soft delete the team (add a record end date), with a confirmation dialog to prevent accidental removal

5. Managing Team Members  
WHEN a BioHub administrator wants to update a team’s membership,  
  THEN they should be able to add or remove users from the team.

## Testing Notes

dwhogan committed:
feat(team): Add team management API endpoints

- GET/POST /api/administrative/teams - List and create teams
- GET/PUT/DELETE /api/administrative/teams/{teamId} - Manage single team
- GET /api/administrative/users - List available users for team membership
- Add TeamRepository pagination and soft-delete filtering
- Add TeamService composite *WithMembers methods
- Add OpenAPI schemas for team endpoints
- Require SYSTEM_ADMIN or DATA_ADMINISTRATOR role

dwhogan committed:
Sonar fix

AntBush reviewed (dismissed):
Looks good!

dwhogan committed:
feat(teams): Add Teams to Users Frontend

dwhogan committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-828

dwhogan committed:
feat(teams): Add server-side search for team member selection

Replace client-side filtering with server-side search to handle large
user bases efficiently. Changes include:

- API: Add search query param with ILIKE filtering and 50 user limit
- Frontend: Implement debounced search (300ms) in TeamMemberSelect
- Cache selected users to persist across search result changes
- Add initialUsers prop to pre-populate when editing existing teams
- Add user icon to dropdown options

mauberti-bc commented on api/src/paths/administrative/users/index.ts:
```diff
@@ -0,0 +1,90 @@
+
+      const search = req.query.search as string | undefined;
+
+      const knex = getKnex();
```
We should keep all sql/DB interactions in the repository layer, like the other endpoints. This index.ts should use the ```userService```:

```
const userService = new UserService(connection)
const users = userService.getAvailableUsers(filters) 
```

And the ```userService.getUsers``` should use the user repository layer:

```
const users = await this.userRepository.getAvailableUsers(filters) 
```

mauberti-bc commented on api/src/services/access-policy/team-service.ts:
```diff
@@ -65,4 +84,133 @@ export class TeamService extends DBService {
+   * @return {Promise<TeamWithMembers>}
+   * @memberof TeamService
+   */
+  async getTeamWithMembers(teamId: string): Promise<TeamWithMembers> {
```
Promise.all here?

mauberti-bc commented on api/src/services/access-policy/team-service.ts:
```diff
@@ -65,4 +84,133 @@ export class TeamService extends DBService {
+  async getTeamMembersWithUsers(teamId: string): Promise<TeamMemberWithUser[]> {
+    const knex = getKnex();
+    const query = knex
+      .table('team_member as tm')
```
Like the first comment: we should only put SQL in the repository layer, so this ```getTeamMembersWithUsers``` should call its counterpart repository method, which can also be called ```getTeamMembersWithUsers```.

mauberti-bc commented:
When the MUI table is set to show 10 rows, the GET request for teams doesn't include limit = 10 for the pagination to work properly. 

In the screenshot, switching to page 2 should make a new request for the next page, which should return a single record.
<img width="397" height="72" alt="image" src="https://github.com/user-attachments/assets/98833c5c-1c9b-45f7-81f4-7cb20e3ea8dc" />

You can find an example of how we handle this in SIMS: https://github.com/bcgov/biohubbc/blob/dev/app/src/features/admin/alert/AlertContainer.tsx. We track paginationModel and sortModel locally, then pass the values to MUI's data grid (see AlertTable.tsx)

dwhogan committed:
fix(team): Move SQL to repository layer and fix pagination

- Refactor: Move all getKnex/SQL from paths/services to repository layer
- Fix: Add server-side pagination/sorting following AlertContainer pattern
- Optimize: Parallelize count/data queries with Promise.all

dwhogan committed:
Merge origin/dev - resolve useApi.ts conflict (keep both search and teams)

dwhogan committed:
fix(app): fixed a couple of sonar issues.

mauberti-bc reviewed (changes requested):
You might still have the ```API_RESPONSE_VALIDATION_ENABLED``` env var false? It should be true to catch these errors. 

<img width="267" height="471" alt="image" src="https://github.com/user-attachments/assets/242b514c-bf8b-4970-91fe-2e56dda29397" />

dwhogan committed:
Merge origin/dev into SIMSBIOHUB-828

dwhogan committed:
fix(api): add profile fields to SystemUser model for merge compatibility

mauberti-bc reviewed (approved):
👍
